ui/src/views/Documentation/Reference/SchemaPanel.jsx

```diff
@@ -0,0 +1,136 @@
+
+@withRouter
+@withStyles(
+  () => ({
+    expansionPanelSummary: {
+      margin: 0,
+    },
+    list: {
+      width: '100%',
+    },
+    gridContainer: {
+      '& > div': {
+        margin: 'auto',
+      },
+    },
+  }),
+  { withTheme: true }
+)
+export default class SchemaPanel extends Component {
+  static propTypes = {
+    /** The reference entry. */
+    entry: object.isRequired,
+    /** The service name in which the entry belongs to. */
+    serviceName: string.isRequired,
+  };
+
+  state = {
+    expanded: false,
+  };
+
+  renderExpansionPanelSummary = () => {
+    const { entry, classes } = this.props;
+
+    return (
+      <Grid className={classes.gridContainer} container spacing={8}>
+        <Grid item xs={5}>
+          <div>
+            <Typography id={entry.content.$id} component="h3">
+              {entry.content.title}
+            </Typography>
+          </div>
+        </Grid>
+        <Grid item xs={7}>
+          <div>
+            <Typography>{entry.content.$id}</Typography>
+          </div>
+        </Grid>
+      </Grid>
+    );
+  };
+
+  renderSchemaTable = (schema, headerTitle) => (
+    <ListItem>
+      <ListItemText
+        primaryTypographyProps={primaryTypographyProps}
+        disableTypography
+        primary={<Typography variant="body1">{headerTitle}</Typography>}
+        secondary={
+          <Fragment>
+            <br />
+            <SchemaTable schema={schema} serviceName="unknown" />
+          </Fragment>
+        }
+      />
+    </ListItem>
+  );
+
+  renderExpansionDetails = () => {
+    const { classes, entry } = this.props;
+    const { expanded } = this.state;
+
+    return (
+      expanded && (
+        <List className={classes.list}>
+          {this.renderSchemaTable(entry.content.$id, entry.content.description)}
+        </List>
+      )
+    );
+  };
+
+  handlePanelChange = key => () => {
+    const { entry, history } = this.props;
+    const { expanded } = this.state;
+
+    if (window.location.hash === `#${entry.content[key]}` || expanded) {
+      history.push(history.location.pathname);
+    } else {
+      history.push(`#${entry.content[key]}`);
+    }
+
+    this.setState({
+      expanded: !expanded,
+    });
+  };
+
+  render() {
+    const { classes } = this.props;
+    const { expanded } = this.state;
+    const entryHashKey = 'content.$id';
+
+    return (
+      <ExpansionPanel
+        defaultExpanded={expanded}
+        onChange={this.handlePanelChange(entryHashKey)}
+        CollapseProps={{ unmountOnExit: true }}>
+        <ExpansionPanelSummary
+          classes={{
+            content: classNames(classes.expansionPanelSummary),
+          }}
+          expandIcon={<ExpandMoreIcon />}>
+          {this.renderExpansionPanelSummary()}
+        </ExpansionPanelSummary>
+        <ExpansionPanelDetails>
+          {this.renderExpansionDetails()}
+        </ExpansionPanelDetails>
+      </ExpansionPanel>
+    );
+  }
+}
```

Comment:
This file seems really similar to `src/views/Documentation/Reference/Entry.js`. Is there a way to make use of that component instead of this file?

Reply:
I think we can pass a type flag to Entry, something like `schema` and use SchemaPanel functions for rendering the SchemaPanel related components.

We might have to do some changes in this logic
```
    const isEntryExchange = type === 'topic-exchange';
    const isLogType = type === 'logs';
    const isFunctionType = !isLogType && !isEntryExchange;
    const entryHashKey = isLogType ? 'type' : 'name';
```

Reply:
Should I change this, to use Entry Component and Remove SchemaPanel component?

Reply:
Yes, please :)